the-collab-lab/tcl-28-smart-shopping-list
# Sd hk firebase

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description



- We installed `firebase` & `react-firebase-hooks`
- We used firestore to add an item in the database. We created our first item, "egg"
- We created 2 components (`ItemsList` and `AddItem`) and imported them to our App
- We created a template for our items. For now, the template is just "name" and "date" but we plan to also include "id" and "frequency", for how often we purchase the item
- We used `useCollection` from `react-firebase-hooks` to display the data onto the UI
- Clicking the "Add an item" button will create a new item in the database

Documentation on `react-firebase-hooks` can be found [here](https://github.com/CSFrequency/react-firebase-hooks/blob/master/firestore/README.md#usecollection)

## Related Issue


#1 

## Acceptance Criteria




## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before




### After


<img width="881" alt="Screen Shot 2021-07-11 at 4 29 21 PM" src="https://user-images.githubusercontent.com/60532744/125209185-2da08180-e265-11eb-98b3-eb0f6e583acd.png">

## Testing Steps / QA Criteria


- From your terminal, pull down this branch
- `npm install`
- `npm start`
- See if data from our `items` collection appears on the app
- Click on the "Add an item" button and a new item should appear
- Check to see if the new item also appears in the `items` collection of our database [here](https://console.firebase.google.com/u/0/project/tcl-28-shopping-list/firestore/data/~2Fitems~2FOoB4YGCWIAZU3qHoXJwz)
